client/components/tinymce/plugins/wpcom-track-paste/plugin.js

```diff
@@ -46,39 +42,38 @@ function pasteTrack( editor ) {
 		return SOURCE_UNKNOWN;
 	}
 
-	function flattenArray( orgTypes ) {
-		// As per https://html.spec.whatwg.org/multipage/interaction.html#datatransfer
-		// types should be an array containing the available formats that were set on copy.
-		// Some browsers, as Firefox, return a DOMStringList instead, see
-		// https://developer.mozilla.org/en-US/docs/Web/API/DOMStringList
-		let types = [];
-		if ( orgTypes.contains ) {
-			types = Array.from( orgTypes );
-		} else {
-			types = orgTypes.slice();
-		}
-		return types.reduce( ( a, b ) => {
-			return a + ', ' + b;
-		} );
+	function toString( types ) {
+		// types should be an array. Some browsers, as Firefox, return a DOMStringList instead.
+		// We need coverage for thoses cases, hence the Array.from conversion.
+		//
+		// https://html.spec.whatwg.org/multipage/interaction.html#datatransfer
+		// DOMStringListhttps://developer.mozilla.org/en-US/docs/Web/API/DOMStringList
+		return Array.from( types ).join( ', ' );
 	}
 
-	function onPasteEditor( event ) {
-		recordEvent( flattenArray( event.clipboardData.types ), 'tinymce' );
+	function onPasteFromTinyMCEEditor( event ) {
+		recordEvent( 'tinymce-editor', toString( event.clipboardData.types ) );
 	}
 
-	function onPasteTextArea( event ) {
-		recordEvent( flattenArray( event.clipboardData.types ), 'html' );
+	function onPasteFromHTMLEditor( event ) {
+		recordEvent( 'html-editor', toString( event.clipboardData.types ) );
 	}
 
-	// Bind paste event listeners to both Visual and HTML editors
-	editor.on( 'paste', onPasteEditor );
+	debug( 'bind listeners for paste event' );
```

Comment:
I won't stop you. just remember that every additional line of code is extra maintenance burden and every line of code not essential to the main thing is distracting us when we read it

:+1: